### PR TITLE
fix of router hanging #2282

### PR DESCRIPTION
**Pull request by askolik**
When generating candidate swaps to perform for the next set of operations to apply, the routing algorithm only checked if there is a candidate swap. It did not check if performing this swap will lead to a new adjacent connection in the graph, considering the max_search_radius. This leads to the algorithm being stuck in an infinite loop for some graph instances, where it keeps performing swaps that don't lead to adjacent connections for the remaining operations.

Fixed this by adding a check for adjacent connections in the new graph, and if there are no new adjacent connections, perform the swap that brings together the farthest nodes.

Fixes: https://github.com/quantumlib/Cirq/issues/2282

### CODE REVIEW

**Comment by askolik** [comment]
@googlebot I fixed it.

File: cirq/contrib/routing/greedy.py [hunk 1]
```diff
@@ -284,7 +284,9 @@ def apply_next_swaps(self):
                 ]
                 if len(candidate_swap_sets) == 1:
                     self.apply_swap(*candidate_swap_sets[0])
-                    return
+
+                    if len(list(self.remaining_dag.findall_nodes_until_blocked(self.acts_on_nonadjacent_qubits))) > 0:
```
Comment (bryano) [inline cirq/contrib/routing/greedy.py]:
```suggestion
                    if list(self.remaining_dag.findall_nodes_until_blocked(self.acts_on_nonadjacent_qubits))):
                        return
                    else:
                        break               
```

Comment (bryano) [reply cirq/contrib/routing/greedy.py]: No reason to keep going through the time slices.

Comment (askolik) [reply cirq/contrib/routing/greedy.py]: You're right, fixed.

**Pull request merged**